main.tf

```diff
@@ -104,16 +104,12 @@ resource "google_compute_firewall" "login_to_vm" {
   count         = (local.create_firewalls && var.allow_login) ? 1 : 0
   name          = local.vm_login_firewall_name
   network       = google_compute_instance.vm_instance.network_interface.0.network
-  source_ranges = [local.google_iap_cidr /* see https://stackoverflow.com/a/57024714/636762 */]
+  source_ranges = concat([local.google_iap_cidr /* see https://stackoverflow.com/a/57024714/636762 */], distinct(var.fw_allowed_cidrs))
```

Comment:
If anyone passes `var.fw_allowed_cidrs = ["35.235.240.0/20"]`, then the result of this would produce `["35.235.240.0/20", "35.235.240.0/20"]` - duplicate CIDR. Instead, we should `concat()` first and THEN `distinct()` over it to de-duplicate these values.

Consider splitting this line into multiple lines for easier readability - this line is now TOO LONG and will require horizontal scrolling in some people's code editors. A good practice is to keep each line within 80-120 characters. Read these [interesting conversations](https://stackoverflow.com/questions/578059/studies-on-optimal-code-width). Lot more in reddit and other places if you google for it.

Consider bringing this line under `target_tag` line to avoid reformatting of too many lines.

Reply:
Done.
Used: 
```
source_ranges = distinct(
    concat(
      [
        local.google_iap_cidr /* see https://stackoverflow.com/a/57024714/636762 */
      ],
      (var.fw_allowed_cidrs)
  ))
```